pkg/webhook/add_webhook.go

```diff
@@ -77,15 +91,50 @@ func newValidatingWebhook(mgr manager.Manager) (*admission.Webhook, error) {
 func newWebhookServer(mgr manager.Manager) (*webhook.Server, error) {
 	return webhook.NewServer("pravega-admission-webhook", mgr, webhook.ServerOptions{
 		CertDir: CertDir,
-		BootstrapOptions: &webhook.BootstrapOptions{
-			// TODO: garbage collect webhook k8s service
-			Service: &webhook.Service{
-				Namespace: os.Getenv("WATCH_NAMESPACE"),
-				Name:      "pravega-admission-webhook",
-				Selectors: map[string]string{
-					"component": "pravega-operator",
+	})
+}
+
+func createWebhookK8sService(mgr manager.Manager) error {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+	c, _ := client.New(cfg, client.Options{Scheme: mgr.GetScheme()})
+	// create webhook k8s service
+	svc := &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pravega-admission-webhook",
+			Namespace: os.Getenv("WATCH_NAMESPACE"),
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"component": "pravega-operator",
+			},
+			Ports: []corev1.ServicePort{
+				{
+					// When using service, kube-apiserver will send admission request to port 443.
+					Port:       443,
+					TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: 443},
 				},
 			},
 		},
-	})
+	}
+	// get operator deployment
+	nn := types.NamespacedName{Namespace: os.Getenv("WATCH_NAMESPACE"), Name: os.Getenv("OPERATOR_NAME")}
```

Comment:
Lets use same namespace as operator, instead of WATCH_NAMESPACE. Operator name also can be obtained from k8sutil.

Reply:
Same as above